agent/engine/task_manager.go

```diff
@@ -25,6 +25,7 @@ import (
 )
 
 const (
+	waitForCredentialsPullTimeout         = 1 * time.Minute
```

Comment:
minor: please rename this to `waitForPullCredentialsTimeout`. Also, this is just a timeout for retrying, yes? I mean if we don't receive creds in `1m`, we'll still retry, yes? Please document that behavior as well.